presto-verifier/src/main/java/com/facebook/presto/verifier/prestoaction/JdbcPrestoAction.java

```diff
@@ -145,7 +146,7 @@ private PrestoConnection getConnection(QueryStage queryStage)
         }
 
         // configure session properties
-        Map<String, String> sessionProperties = queryStage.isMain()
+        Map<String, String> sessionProperties = queryStage.isMain() || queryStage == DETERMINISM_ANALYSIS_MAIN
```

Comment:
Is there any reason not to include DETERMINISM_ANALYSIS_MAIN into isMain?

Reply:
There are other callers of `isMain()` that expect the query stage to be the control/test main query, but not determinism analysis stages.